src/main/java/org/skriptlang/skript/bukkit/toolcomponent/elements/ExprToolCompDamage.java

```diff
@@ -0,0 +1,127 @@
+package org.skriptlang.skript.bukkit.toolcomponent.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.bukkitutil.ItemUtils;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.slot.Slot;
+import ch.njol.util.Kleenean;
+import ch.njol.util.Math2;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.components.ToolComponent;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.function.Consumer;
+
+@Name("Tool Component - Damage Per Block")
+@Description("The damage the tool receives when a block is broken.")
+@Examples({
+	"set the damage per block of {_tool} to 10",
+	"",
+	"set {_component} to the tool component of {_tool}",
+	"set the damage per block of {_component} to 5",
+	"set the tool component of {_tool} to {_component}"
+})
+@RequiredPlugins("Minecraft 1.20.6+")
+@Since("INSERT VERSION")
+public class ExprToolCompDamage extends PropertyExpression<Object, Number> {
+
+	static {
+		Skript.registerExpression(ExprToolCompMiningSpeed.class, Number.class, ExpressionType.PROPERTY,
```

Comment:
Confused on the random support of itemtype, itemstack and slot

Reply:
Allowing itemtypes, itemstacks and slots allows direct manipulation to the items rather than having it only be to a tool component then applying that tool component to multiple items. Especially if the multiple items have different data for their tool components.